bip-xxxx.mediawiki

```diff
@@ -0,0 +1,303 @@
+Small change inside wallets are detrimental to privacy. Mixers like Wasabi wallet, because of its protocol, eventually generate such [[https://docs.wasabiwallet.io/using-wasabi/ChangeCoins.html#spend-the-change-with-another-entity-where-you-don-t-mind-if-each-of-the-two-knows-that-you-transact-with-the-other-entity|small change]].
+
+A common way to protect your privacy is to donate those spare changes, to deposit them in an exchange or on your favorite merchant's store account. Those kind of transactions can easily be spotted on the blockchain: There is only one output.
+
+However, if you donate via payjoin, it will look like a normal transaction.
+
+On top of this the receiver can poison analysis by randomly faking a round amount of satoshi for the additional output.
+
+===Impacted heuristics===
+Our proposal of payjoin is breaking the following blockchain heuristics:
+
+* Common inputs heuristics.
+
+Because payjoin is mixing the inputs of the sender and receiver, this heuristic becomes unreliable.
+
+* Change identification from scriptPubKey type heuristics.
+
+When Alice pays Bob, if Alice is using P2SH but Bob's deposit address is P2WPKH, the heuristic would assume that the P2SH output is the change address of Alice.
+This is now however a broken assumption, as the payjoin receiver has the freedom to mislead analytics by purposefully changing the invoice's address in the payjoin transaction.
+
+If Bob's original address is P2WPKH and Alice is sending from P2SH, then Bob can change the receiving address in the payjoin transaction to be P2SH, preventing the use of this heuristic.
+
+Alternatively, if the original address of Bob is P2WPKH and Alice is also P2WPKH, Bob can change the receiving address in the payjoin to P2SH. The heuristic would wrongfully identify the payjoin's receiving address as the change address of the transaction.
+
+* Change identification from round change amount
+
+If Alice pays Bob, she might be tempted to pay him a round amount, like <code>1.23000000 BTC</code>. When this happens, blockchain analysis often identify the output without the round amount as the change of the transaction.
+
+For this reason, during a [spare change](Payjoin-spec.md#spare-change-donation) situation, we randomly round the amount in the output added by the receiver to the payjoin transaction.
+
+==Attack vectors==
+
+===On the receiver side: UTXO probing attack===
+
+When the receiver creates a payjoin proposal, they expose one or more inputs belonging to them.
+
+An attacker could create multiple original transactions in order to learn the UTXOs of the receiver, while not broadcasting the payjoin proposal.
+
+While we cannot prevent this type of attack entirely, we implemented the following mitigations:
+
+* When the receiver detects an original transaction being broadcasted, or if the receiver detects that the original transaction has been double spent, then they will reuse the UTXO that was exposed for the next payjoin.
```

Comment:
Should the receiver accept payjoin request with `nLockTime` in the future? It could be easy to scan the receiver utxo by sending transactions that can only be mined in 100 years.

Reply:
It can't because the receiver and the sender are broadcasting the original PSBT after 2 minutes whatever happen.

Reply:
@NicolasDorier you have answer this same question twice to me but I still don't get it. Do you say the receiver can broadcast a transaction that cannot be mined?

Reply:
Sorry I misunderstood. I thought you were talking about an invalid payjoin tx (in which case, this is not a problem as the original transaction will be broadcasted).

In BTCPay Server, we verify that the transaction is broadcastable.
For Wasabi wallet, the sender will not be able to automatically retry a payjoin transaction so he would only learn about a single utxo.